packages/create-plugin/src/utils/utils.templates.ts

```diff
@@ -88,23 +92,70 @@ export function renderTemplateFromFile(templateFile: string, data?: any) {
   return renderHandlebarsTemplate(fs.readFileSync(templateFile).toString(), data);
 }
 
-export function getTemplateData(): TemplateData {
+export function buildProxyFromJson() {
   const pluginJson = getPluginJson();
+  return new Proxy(pluginJson, {
+    get(target, prop: string | symbol) {
+      if (prop === 'pluginId') {
+        return target.id;
+      } else if (prop === 'pluginName') {
+        return target.name;
+      } else if (prop === 'pluginDescription') {
+        return target.info?.description;
+      } else if (prop === 'hasBackend') {
+        return target.backend;
+      } else if (prop === 'hasGithubWorkflows') {
+        return undefined;
+      } else if (prop === 'hasGithubLevitateWorkflow') {
+        return undefined;
+      } else if (prop === 'orgName') {
+        return target.info?.author?.name;
+      } else if (prop === 'pluginType') {
+        return target.type;
+      } else if (prop === 'packageManager') {
+        return getPackageManagerWithFallback();
+      } else {
+        throw new Error(`Property ${String(prop)} not found in package.json.`);
+      }
+    },
+  });
+}
+export function buildProxyFromUserPrompt(target: CliArgs) {
+  return new Proxy(target, {
```

Comment:
As above, I'm not sure what the benefits are of using a proxy here.

Reply:
(Resolving this, as the proxy got removed.)